__tests__/app.test.js

```diff
@@ -164,5 +159,253 @@ describe("post/api/articles/:articleid/comments", () => {
+    return request(app)
+      .get("/api/articles?sort_by=break")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("invalid sort_by query");
+      });
+  });
+  test("invalid order parameters cause error", () => {
+    return request(app)
+      .get("/api/articles?order=break")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("invalid order query");
+      });
+  });
+  test("invalid order parameters cause error", () => {
+    return request(app)
+      .get("/api/articles?order=break")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("invalid order query");
+      });
+  });
+  test("topic parameters for topic that does not exist causes 404 error", () => {
+    return request(app)
+      .get("/api/articles?topic=break")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("No data for this topic");
+      });
+  });
+});
+
+describe("get/api/articles/:articleid/comments?queries", () => {
+  test("now returns comment count aswell if given the query count", () => {
+    return request(app)
+      .get("/api/articles/3/comments?count=true")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body).toEqual([
+          [
+            {
+              body: "git push origin master",
+              votes: 0,
+              author: "icellusedkars",
+              comment_id: 10,
+              article_id: 3,
+              created_at: "2020-06-20T07:24:00.000Z",
+            },
+            {
+              body: "Ambidextrous marsupial",
+              votes: 0,
+              author: "icellusedkars",
+              comment_id: 11,
+              article_id: 3,
+              created_at: "2020-09-19T23:10:00.000Z",
+            },
+          ],
+          { comment_count: 2 },
+        ]);
+      });
+  });
+  test("if query is not specified as count = true, it is assumed false", () => {
+    return request(app)
+      .get("/api/articles/3/comments?count=yes")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body).toEqual([
+          {
+            body: "git push origin master",
+            votes: 0,
+            author: "icellusedkars",
+            comment_id: 10,
+            article_id: 3,
+            created_at: "2020-06-20T07:24:00.000Z",
+          },
+          {
+            body: "Ambidextrous marsupial",
+            votes: 0,
+            author: "icellusedkars",
+            comment_id: 11,
+            article_id: 3,
+            created_at: "2020-09-19T23:10:00.000Z",
+          },
+        ]);
+      });
+  });
+});
+describe("delete /api/comments/:comment_id", () => {
+  test("deletes the comment matching the given ID, gives 404 when ran again as comment is now deleted", () => {
+    return request(app)
+      .delete("/api/comments/3")
+      .expect(204)
+      .then(() => {
+        return request(app).delete("/api/comments/3").expect(404);
```

Comment:
12: This is a nice way of checking that our change has been committed to the database. You could also use your GET comment by id endpoint to achieve the same thing.